xapian-applications/omega/scriptindex.cc

```diff
@@ -495,7 +496,11 @@ parse_index_script(const string &filename)
 			// We don't push an Action for WEIGHT - instead we
 			// store it ready to use in the INDEX and INDEXNOPOS
 			// Actions.
-			weight = atoi(val.c_str());
+			if (!parse_signed(val.c_str(), weight)) {
```

Comment:
Here weight should really be non-negative (it's how much the `wdf` is increased by), though the current docs do just say "an integer".  It would be a bit odd to specify zero, but I can see there might be uses for indexing text without increasing wdf.

Reply:
fixed in a fixup commit.